Makefile

```diff
@@ -81,13 +81,29 @@ paas-login: ## Log in to PaaS
 	$(if ${PAAS_SPACE},,$(error Must specify PAAS_SPACE))
 	@cf login -a "${PAAS_API}" -u ${PAAS_USERNAME} -p "${PAAS_PASSWORD}" -o "${PAAS_ORG}" -s "${PAAS_SPACE}"
 
+.PHONY: add-all-app-network-policies
+add-all-app-network-policies: ## attempts to (re-)add all known PaaS "network policies" suitable for a particular $STAGE
+	$(call check_space)
+	$(if ${STAGE},,$(error Must specify STAGE))
+	for APPLICATION_NAME in $$(yq -rs '.[0] * .[1] | to_entries | .[] | select(.value | has("egress_to_applications"))? | .key' ./vars/common.yml ./vars/${STAGE}.yml) ; do \
+		./scripts/add-application-network-policies.sh ./vars/$${STAGE}.yml ./vars/common.yml $${APPLICATION_NAME} ; \
+	done
+
 .PHONY: deploy-app
 deploy-app: ## Deploys the app to PaaS
 	$(call check_space)
 	$(if ${APPLICATION_NAME},,$(error Must specify APPLICATION_NAME))
 	$(if ${RELEASE_NAME},,$(error Must specify RELEASE_NAME))
+	$(if ${STAGE},,$(error Must specify STAGE))
 	cf push --no-start -f <(make -s -C ${CURDIR} generate-manifest) -o digitalmarketplace/${APPLICATION_NAME}:${RELEASE_NAME}
 
+	# the order of operations performed here (including inside add-application-network-policies.sh)
```

Comment:
This comment doesn't give us much info on the _why_ it's done here. Just that it's important it is done here.

Something like:
# This step must be performed after XYZ but before ABC to ensure DEF

Would help other devs get where they need to be quicker

Reply:
I'll take a look and try, but the exact reason for it will probably have a long long explanation for it - it's not something I've actually thought about in terms of english language, it's just that I found this order of operations impossible to trip up no matter where it gets interrupted...

Reply:
See, the way I determined the validity of this scheme was really by playing a sort of card game in my text editor trying to find different interleaving orders that would end up starting an app without its policies in place:
![screen1](https://user-images.githubusercontent.com/807447/53799467-55afa580-3f32-11e9-93e4-e865a4c11783.png)

Reply:
In fact, the following is a particularly interesting case, because from B's POV, both of its attempted policy additions would have appeared to fail, but actually the end result would end up being fine because _both_ of A's additions of them should have succeeded.
![screen1](https://user-images.githubusercontent.com/807447/53800172-05394780-3f34-11e9-8ccc-ea76479172c7.png)

I _do_ think though I can make the failure messages a bit more illustrative of this.